buildSrc/src/main/groovy/net/minecraftforge/gradleutils/GradleUtils.groovy

```diff
@@ -84,24 +84,68 @@ class GradleUtils {
      * Get a closure to be passed into {@link org.gradle.api.artifacts.dsl.RepositoryHandler#maven(groovy.lang.Closure)}
      * in a publishing block.
      *
+     * Important the following environment variables must be set for this to work:
+     *  - MAVEN_USER: Containing the username to use for authentication
+     *  - MAVEN_PASSWORD: Containing the password to use for authentication
+     *  - MAVEN_URL_RELEASE: Containing the URL to use for the release repository
+     *  - MAVEN_URL_SNAPSHOT: Containing the URL to use for the snapshot repository
+     *
      * @param project The project
      * @param defaultFolder The default folder if the required maven information is not currently set
      * @return a closure
      */
     static getPublishingForgeMaven(Project project, File defaultFolder = project.rootProject.file('repo')) {
+        return setupSnapshotCompatiblePublishing(project, 'https://maven.minecraftforge.net/', defaultFolder);
+    }
+
+    /**
+     * Get a closure to be passed into {@link org.gradle.api.artifacts.dsl.RepositoryHandler#maven(groovy.lang.Closure)}
+     * in a publishing block, this closure respects the current project's version, with regards to publishing to a release
+     * or snapshot repository.
+     *
+     * Important the following environment variables must be set for this to work:
+     *  - MAVEN_USER: Containing the username to use for authentication
+     *  - MAVEN_PASSWORD: Containing the password to use for authentication
+     *
+     * The following environment variables are optional:
+     *  - MAVEN_URL_RELEASE: Containing the URL to use for the release repository
+     *  - MAVEN_URL_SNAPSHOT: Containing the URL to use for the snapshot repository
+     *
+     * If the MAVEN_URL_RELEASE is not set the passed in fallback URL will be used for the release repository.
+     * By default this is: https://maven.minecraftforge.net/
+     * This is done to preserve backwards compatibility with the old {@link #getPublishingForgeMaven(Project, File)} method.
+     *
+     * @param project The project
+     * @param defaultFolder The default folder if the required maven information is not currently set
+     * @return a closure
+     */
+    static setupSnapshotCompatiblePublishing(Project project, String fallbackPublishingEndpoint = 'https://maven.minecraftforge.net/', File defaultFolder = project.rootProject.file('repo'), File defaultSnapshotFolder = project.rootProject.file('snapshots')) {
         return { MavenArtifactRepository it ->
             name 'forge'
-            if (System.env.MAVEN_USER) {
-                url 'https://maven.minecraftforge.net/'
+            if (System.env.MAVEN_USER && System.env.MAVEN_PASSWORD) {
+                def publishingEndpoint = fallbackPublishingEndpoint;
+                if (System.env.MAVEN_URL_RELEASES) {
+                    publishingEndpoint = System.env.MAVEN_URL_RELEASES;
```

Comment:
The javadocs mention `MAVEN_URL_RELEASE`, while you use `MAVEN_URL_RELEASES` here. Which one is it? These semicolons are also unnecessary

Reply:
it should be RELEASE not RELEASES.
I fixed this now.
Good catch.